src/parser/SpatialQuery.cpp

```diff
@@ -0,0 +1,149 @@
+  } else if (predString.ends_with("nearestNeighbors>")) {
+    if (!object.isInt()) {
+      throw SpatialSearchException(
+          "The parameter 'nearestNeighbors' expects an integer (the maximum "
+          "number of nearest neighbors)");
+    }
+    maxResults_ = object.getInt();
+  } else if (predString.ends_with("maxDistance>")) {
+    if (!object.isInt()) {
+      throw SpatialSearchException(
+          "The parameter 'maxDistance' expects an integer (the maximum "
+          "distance in meters)");
+    }
+    maxDist_ = object.getInt();
+  } else if (predString.ends_with("bindDistance>")) {
+    setVariable("bindDistance", object, bindDist_);
+  } else if (predString.ends_with("algorithm>")) {
+    if (!object.isIri()) {
+      throw SpatialSearchException(
+          "The parameter 'algorithm' needs an IRI that selects the algorithm "
+          "to employ. Currently supported are 'spatialSearch:baseline' and "
+          "'spatialSearch:s2'.");
+    }
+    auto algoIri = object.getIri().toStringRepresentation();
+    if (algoIri.ends_with("baseline>")) {
+      algo_ = SpatialJoinAlgorithm::BASELINE;
+    } else if (algoIri.ends_with("s2>")) {
+      algo_ = SpatialJoinAlgorithm::S2_GEOMETRY;
+    } else {
+      throw SpatialSearchException(
+          "The IRI given for the parameter 'algorithm' does not refer to a "
+          "supported spatial search algorithm. Please select either "
+          "'spatialSearch:baseline' or 'spatialSearch:s2'.");
+    }
+  } else {
+    throw SpatialSearchException(
+        "Unsupported argument " + predString +
+        " in Spatial Search. Supported Arguments: left, right, "
+        "nearestNeighbors, maxDistance, bindDistance and algorithm.");
+  }
+}
+
+// ____________________________________________________________________________
+SpatialJoinConfiguration SpatialQuery::toSpatialJoinConfiguration() const {
+  if (!left_.has_value()) {
+    throw SpatialSearchException("Missing parameter 'left' in spatial search.");
+  } else if (!maxDist_.has_value() && !maxResults_.has_value()) {
+    throw SpatialSearchException(
+        "Neither 'nearestNeighbors' nor 'maxDistance' were provided. At least "
+        "one of both is required.");
+  } else if (!right_.has_value()) {
+    throw SpatialSearchException(
+        "Missing parameter 'right' in spatial search.");
+  }
+
+  // Only if the number of results is limited, it is mandatory that the right
+  // variable must be selected inside the service. If only the distance is
+  // limited, it may be declared inside or outside of the service.
+  if (maxResults_.has_value() && childGraphPattern_._graphPatterns.empty()) {
+    throw SpatialSearchException(
+        "Missing parameter 'right' in spatial search. A spatial search with "
+        "a maximum number of results must have its right variable declared "
+        "inside the service using a graph pattern: SERVICE spatialSearch: { "
+        "... { ... ?right } }.");
+  }
```

Comment:
This is misleading. You mean that in this case you need the GroupGraphPattern which specifies the right input, the `<right> `Variable was probably specified.